File: Examples/DeveloperObjcExample/ObjcExample/SignInViewController.m
//
//  SignInViewController.m
//  AppcuesObjcExample
//
//  Created by Matt on 2022-06-01.
//

#import "SignInViewController.h"
#import "Appcues+Shared.h"

NSString *currentUserID = @"default-00000";

@interface SignInViewController ()
@property (weak, nonatomic) IBOutlet UITextField *userIDTextField;

@end

@implementation SignInViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    _userIDTextField.text = currentUserID;
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    [[Appcues shared] screenWithTitle:@"Sign In" properties:nil];
}

- (IBAction)signInTapped:(UIButton *)sender {
    NSString *userID = _userIDTextField.text;

    [[Appcues shared] identifyWithUserID:userID properties:nil];

    currentUserID = userID;
}

- (IBAction)signOutAction:(UIStoryboardSegue *)unwindSegue {
    // Unwind to Sign In
    [[Appcues shared] reset];
}

- (IBAction)anonymousUserTapped:(UIButton *)sender {
    [[Appcues shared] anonymousWithProperties:nil];
}

@end

File: Examples/DeveloperCocoapodsExample/CocoapodsExample/SignInViewController.swift
//
//  SignInViewController.swift
//  CocoapodsExample
//
//  Created by Matt on 2021-10-12.
//

import UIKit
import AppcuesKit

enum User {
    static var currentID = "default-00000"
}

class SignInViewController: UIViewController {

    private var pushStatus = "unknown"

    @IBOutlet private var userIDTextField: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()

        // determine push status
        let current = UNUserNotificationCenter.current()
        current.getNotificationSettings { settings in
            if settings.authorizationStatus == .notDetermined {
                // Notification permission has not been asked yet, go for it!
                self.pushStatus = "notDetermined"
            } else if settings.authorizationStatus == .denied {
                // Notification permission was previously denied, go to settings & privacy to re-enable
                self.pushStatus = "denied"
            } else if settings.authorizationStatus == .authorized {
                // Notification permission was already granted
                self.pushStatus = "authorized"
            }
        }

        userIDTextField.text = User.currentID
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        Appcues.shared.screen(title: "Sign In")
    }

    @IBAction private func signInTapped(_ sender: UIButton) {
        let userID = userIDTextField.text ?? User.currentID

        Appcues.shared.identify(userID: userID, properties: ["pushStatus": pushStatus])

        User.currentID = userID
    }

    @IBAction private func signOutAction(unwindSegue: UIStoryboardSegue) {
        // Unwind to Sign In
        Appcues.shared.reset()
    }

    @IBAction private func anonymousUserTapped(_ sender: Any) {
        Appcues.shared.anonymous()
    }
}
